pkg/executor/join/hash_join_v2.go

```diff
@@ -588,40 +524,31 @@ func (w *ProbeWorkerV2) getNewJoinResult() (bool, *hashjoinWorkerResult) {
 func (e *HashJoinV2Exec) Next(ctx context.Context, req *chunk.Chunk) (err error) {
 	if !e.prepared {
 		e.initHashTableContext()
+		e.initializeForProbe()
 		e.hashTableContext.memoryTracker.AttachTo(e.memTracker)
 		e.buildFinished = make(chan error, 1)
-		e.workerWg.RunWithRecover(func() {
-			defer trace.StartRegion(ctx, "HashJoinHashTableBuilder").End()
-			e.fetchAndBuildHashTable(ctx)
-		}, e.handleFetchAndBuildHashTablePanic)
+		e.fetchAndBuildHashTable(ctx)
 		e.fetchAndProbeHashTable(ctx)
 		e.prepared = true
 	}
-	if e.ProbeSideTupleFetcher.shouldLimitProbeFetchSize() {
-		atomic.StoreInt64(&e.ProbeSideTupleFetcher.requiredRows, int64(req.RequiredRows()))
-	}
+
+	e.updateRequiredRows(req)
 	req.Reset()
 
 	result, ok := <-e.joinResultCh
+	e.recycleChunk(req, result)
```

Comment:
why move this recycle code before error check?

Reply:
> why move this recycle code before error check?

If error happens, we will directly return and fail to recycle chunk, this will let workers hang.